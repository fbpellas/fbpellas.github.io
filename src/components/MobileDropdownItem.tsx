import * as React from 'react';
import NavDropdown from 'react-bootstrap/NavDropdown';

export interface MobileDropdownItemProps {
  anchor: string;
  setPageAndClear: (hash: string) => void;
  title: string;
}

const MobileDropdownItem: React.FC<MobileDropdownItemProps> = (props) => {
  const { anchor, setPageAndClear, title } = props;
  const href = `#${anchor}`;

  return (
    <NavDropdown.Item
      href={href}
      onClick={() => {
        setPageAndClear(anchor);
      }}
    >
      {title}
    </NavDropdown.Item>
  );
};

export { MobileDropdownItem };
