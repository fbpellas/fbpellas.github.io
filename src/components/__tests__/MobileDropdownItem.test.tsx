import React from 'react';
import { render } from '@testing-library/react';
import { MobileDropdownItem } from '../MobileDropdownItem';

test('MobileDropdownItem', () => {
  const fn = jest.fn();
  const { container } = render(<MobileDropdownItem anchor="some-anchor" setPageAndClear={fn} title="Some Title" />);
  expect(container).toMatchSnapshot();
});
