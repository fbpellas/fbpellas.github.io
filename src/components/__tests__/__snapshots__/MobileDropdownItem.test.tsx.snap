// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileDropdownItem 1`] = `
<div>
  <a
    class="dropdown-item"
    href="#some-anchor"
  >
    Some Title
  </a>
</div>
`;
